src/btree/bt_read.c

```diff
@@ -125,17 +128,258 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+                if (cmp == 0) {
+                    WT_ERR(__create_birthmark_upd(session, birthmarkp, &upd));
+                    birthmark_record = true;
+                    break;
+                } else if (cmp < 0)
+                    /* LAS key is less than birthmark key, so no birthmark record for that key */
+                    break;
+                else
+                    /* LAS key is greater than birthmark key, check next set of birthmark records */
+                    continue;
+            }
+        }
+
+        if (upd == NULL) {
+            /* Allocate the WT_UPDATE structure. */
+            WT_ERR(
+              cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+
+            WT_ERR(__wt_update_alloc(session, &las_value, &upd, &incr, upd_type));
+            upd->txnid = las_txnid;
+            upd->durable_ts = durable_timestamp;
+            upd->start_ts = las_timestamp;
+            upd->prepare_state = prepare_state;
+        }
+
+        switch (page->type) {
+        case WT_PAGE_COL_FIX:
+        case WT_PAGE_COL_VAR:
+            p = las_key.data;
+            WT_ERR(__wt_vunpack_uint(&p, 0, &recno));
+            WT_ERR(__col_instantiate(session, recno, ref, &cbt, upd));
+            break;
+        case WT_PAGE_ROW_LEAF:
+            WT_ERR(__row_instantiate(session, &las_key, ref, &cbt, upd));
+            break;
+        default:
+            WT_ERR(__wt_illegal_value(session, page->type));
+        }
+
+        /* Remove the prepared record from LAS once the page is instantiated successfully */
+        if (!birthmark_record && upd->prepare_state == WT_PREPARE_INPROGRESS) {
+            /* Extend the buffer if needed */
+            WT_ERR(__wt_buf_extend(session, las_prepares,
+              (las_prepare_cnt + 1) * sizeof(struct las_page_prepared_updates)));
+            las_preparep = (struct las_page_prepared_updates *)las_prepares->mem + las_prepare_cnt;
+            WT_CLEAR(las_preparep->key);
+            WT_ERR(__wt_buf_set(session, &las_preparep->key, las_key.data, las_key.size));
+            las_preparep->timestamp = las_timestamp;
+            las_preparep->txnid = las_txnid;
+            las_prepare_cnt++;
+        }
+
+        if (birthmark_record) {
+            birthmarkp->instantiated = true;
+            birthmark_record = false;
+        }
+
+        upd = NULL;
+    }
+
+    __wt_readunlock(session, &cache->las_sweepwalk_lock);
+    locked = false;
+    WT_ERR_NOTFOUND_OK(ret);
+
+    __wt_verbose(session, WT_VERB_LOOKASIDE_ACTIVITY,
+      "Page is instantiated with LAS content for btree ID %" PRIu32
+      " with number of keys: %" PRId32,
+      las_btree_id, instantiated_cnt);
+
+    /* Instantiate the remaining birthmark entries */
+    WT_ERR(__instantiate_birthmarks(session, ref));
+
+    /*
+     * Now the page is successfully instantiated. It is unlikely that removing of prepared updates
+     * to fail.
```

Comment:
Please reword the second sentence to: `It is unlikely that removing the prepared updates will fail.`

I'm not sure of the intent of that comment actually. Most error paths are unlikely, but possible. Is there something unusual about this situation? Is there a problem or special handling needed if we error at this point?

Reply:
Changed the comment, this is not a special error scenario.